Stellar-contract-config-setting.x

```diff
@@ -153,6 +153,25 @@ struct StateExpirationSettings {
     int64 restorableRentRateDenominator;
     int64 tempRentRateDenominator;
 
+    // max number of entries that emit expiration meta in a single ledger
+    uint32 maxEntriesToExpire;
+
+    // maximum bytes to scan the BucketList for expired entries in a single ledger, in megabytes
+    uint32 maxBytesToScan;
+
+    union switch (int v)
```

Comment:
For extensions please use `ExtensionPoint` and put it at the top of the entry.

Also, config extensions are kind of unnecessary, aren't they? You could always just add one more entry type (that's why most of the entries aren't extensible).

Reply:
> For extensions please use ExtensionPoint and put it at the top of the entry.

I've seen all the ExtensionPoints moved to top of the entry and been wondering. Is it just stylistic or are their other reasons (such as maintaining a certain sorting order)? Most of the pre-existing entries have `ext` as the last field.

Reply:
@jayz22 We started placing extension points at the beginning because it defers the choice of:
- extension points being dangling where v2 extends off v1
- or, new versions being an entire replacement since it'll allow us to change the type to be a union in a binary compatible way

Reply:
Hmm, still not sure I understand why putting it at the beginning instead of at the end helps with deferring the choice.

Reply:
The main reason would be the possibility of converting extension point to union.